Passing the filename from the sender device to the reciever device.

Trevato opened the pull request:
Now when a file is passed it will take the same name that it had originally from the sender. This will enable any filetype to be passed.

Trevato committed:
Passing the filename from the sender device to the reciever device.

Now when a file is passed it will take the same name that it had originally from the sender. This will enable
any filetype to be passed.

lauzadis committed:
Fixed unicode characters. Still broken filename

lauzadis commented:
The SendActivity needs to be modified with the fixed null terminator unicode. I recommend using 0.tochar() or '\u0000'

Trevato committed:
Fix filename transfer.

Filename is now sent across devices. We can use some of this logic to change headers back to raw Bytes.

Trevato committed:
Allowing reciever to pick file location with the FilePicker.

Working on adding a file picker for the recieving user. This allows them to choose where the new file goes.
Also removed a lot of unused functions and generally cleaned up the code.
